autogpt_platform/backend/backend/server/v2/postmark/models.py

```diff
@@ -0,0 +1,212 @@
+    """
+    Registration not confirmed — The subscriber has not clicked on the confirmation link upon registration or import.
+    """
+    Blocked = 100006
+    """
+    Blocked from this ISP due to content or blacklisting.
+    """
+    SMTPApiError = 100007
+    """
+    An error occurred while accepting an email through the SMTP API.
+    """
+    InboundError = 100008
+    """
+    Processing failed — Unable to deliver inbound message to destination inbound hook.
+    """
+    DMARCPolicy = 100009
+    """
+    Email rejected due DMARC Policy.
+    """
+    TemplateRenderingFailed = 100010
+    """
+    Template rendering failed — An error occurred while attempting to render your template.
+    """
+
+
+class PostmarkBounceWebhook(BaseModel):
+    RecordType: Literal["Bounce"] = "Bounce"
+    ID: int
+    Type: str
+    TypeCode: PostmarkBounceEnum
+    Tag: str
+    MessageID: str
+    Details: str
+    Email: str
+    From: str
+    BouncedAt: str
+    Inactive: bool
+    DumpAvailable: bool
+    CanActivate: bool
+    Subject: str
+    ServerID: int
+    MessageStream: str
+    Content: str
+    Name: str
+    Description: str
+    Metadata: dict[str, str]
+
+
+class PostmarkSpamComplaintWebhook(BaseModel):
+    RecordType: Literal["SpamComplaint"] = "SpamComplaint"
+    ID: int
+    Type: str
+    TypeCode: int
+    Tag: str
+    MessageID: str
+    Details: str
+    Email: str
+    From: str
+    BouncedAt: str
+    Inactive: bool
+    DumpAvailable: bool
+    CanActivate: bool
+    Subject: str
+    ServerID: int
+    MessageStream: str
+    Content: str
+    Name: str
+    Description: str
+    Metadata: dict[str, str]
+
+
+class PostmarkOpenWebhook(BaseModel):
+    RecordType: Literal["Open"] = "Open"
+    MessageStream: str
+    Metadata: dict[str, str]
+    FirstOpen: bool
+    Recipient: str
+    MessageID: str
+    ReceivedAt: str
+    Platform: str
+    ReadSeconds: int
+    Tag: str
+    UserAgent: str
+    OS: dict[str, str]
+    Client: dict[str, str]
+    Geo: dict[str, str]
+
+
+class PostmarkClickWebhook(BaseModel):
+    RecordType: Literal["Click"] = "Click"
+    MessageStream: str
+    Metadata: dict[str, str]
+    Recipient: str
+    MessageID: str
+    ReceivedAt: str
+    Platform: str
+    ClickLocation: str
+    OriginalLink: str
+    Tag: str
+    UserAgent: str
+    OS: dict[str, str]
+    Client: dict[str, str]
+    Geo: dict[str, str]
+
+
+class PostmarkSubscriptionChangeWebhook(BaseModel):
+    RecordType: Literal["SubscriptionChange"] = "SubscriptionChange"
+    MessageID: str
+    ServerID: int
+    MessageStream: str
+    ChangedAt: str
+    Recipient: str
+    Origin: str
+    SuppressSending: bool
+    SuppressionReason: str
+    Tag: str
+    Metadata: dict[str, str]
+
+
+PostmarkWebhook = (
+    PostmarkDeliveryWebhook
+    | PostmarkBounceWebhook
+    | PostmarkSpamComplaintWebhook
+    | PostmarkOpenWebhook
+    | PostmarkClickWebhook
+    | PostmarkSubscriptionChangeWebhook
+)
```

Comment:
Is this auto-generated? How do we keep this up to date or at least know something is breaking with our library being auto-updated.

Reply:
We don't and its not something we can manage :(